component/azstorage/config.go

```diff
@@ -391,13 +391,22 @@ func ParseAndValidateConfig(az *AzStorage, opt AzStorageOptions) error {
 		az.stConfig.authConfig.SASKey = sanitizeSASKey(opt.SaSKey)
 	case EAuthType.MSI():
 		az.stConfig.authConfig.AuthMode = EAuthType.MSI()
-		if opt.ApplicationID == "" && opt.ResourceID == "" {
-			//lint:ignore ST1005 ignore
-			return errors.New("Application ID and Resource ID not provided")
+		v := make(map[string]bool, 3)
+		if opt.ApplicationID != "" {
+			v[opt.ApplicationID] = true
+		}
+		if opt.ObjectID != "" {
+			v[opt.ObjectID] = true
+		}
+		if opt.ResourceID != "" {
+			v[opt.ResourceID] = true
+		}
+		if len(v) != 1 {
```

Comment:
This should be `> 1` instead of `!= 1`. Not specifying any of the three values (AppId, ObjId, or ResId) is also a valid scenario for MSI with System assigned identity. In that case, the IMDS endpoint gets a token for VM's identity. 

Here is a [reference](https://learn.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/managed-identities-faq#what-identity-will-imds-default-to-if-dont-specify-the-identity-in-the-request). For example [oAuthTokenManager.go](https://github.com/Azure/azure-storage-azcopy/blob/main/common/oauthTokenManager.go#L656)